var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -744,119 +742,95 @@ def libs(self):
 
     @run_after("install")
     @on_package_attributes(run_tests=True)
-    def test_install(self):
-        """Perform make test_install."""
-        with working_dir(self.build_directory):
+    def check_test_install(self):
+        """Perform test_install on the build."""
+        with working_dir(self.builder.build_directory):
             make("test_install")
 
     @property
-    def _smoke_tests(self):
-        # smoke_tests tuple: exe, args, purpose, use cmake (true/false)
-        smoke_tests = [
-            ("nvector/serial/test_nvector_serial", ["10", "0"], "Test serial N_Vector", False)
-        ]
-        if "+CVODE" in self.spec:
-            smoke_tests.append(("cvode/serial/cvAdvDiff_bnd", [], "Test CVODE", True))
+    def _smoke_tests_path(self):
+        # examples/smoke-tests are cached for testing
+        return self.prefix.examples
 
-        if "+cuda" in self.spec:
-            smoke_tests.append(
-                ("nvector/cuda/test_nvector_cuda", ["10", "0", "0"], "Test CUDA N_Vector", True)
-            )
-            if "+CVODE" in self.spec:
-                smoke_tests.append(
-                    ("cvode/cuda/cvAdvDiff_kry_cuda", [], "Test CVODE with CUDA", True)
-                )
+    def run_example(self, exe_path, opts, cmake_bool):
+        """Common sundials test method"""
+        if "~examples-install" in self.spec:
+            raise SkipTest("Package must be installed with +examples-install")
 
-        if "+rocm" in self.spec:
-            smoke_tests.append(
-                ("nvector/hip/test_nvector_hip", ["10", "0", "0"], "Test HIP N_Vector", True)
-            )
-            if "+CVODE" in self.spec:
-                smoke_tests.append(
-                    ("cvode/hip/cvAdvDiff_kry_hip", [], "Test CVODE with HIP", True)
-                )
+        (dirname, basename) = os.path.split(exe_path)
+        srcpath = join_path(self._smoke_tests_path, dirname)
+        if not os.path.exists(srcpath):
+            raise SkipTest(f"Example '{dirname}' not found in {self.version}")
```

Comment:
Shouldn't this be `basename` instead of `dirname`?

Reply:
I debated on that one.

If you look at the original and current entries, `basename` is the name of the executable that hasn't been built at this point.  I was trying to avoid the overhead of the context, `Executable`s instantiation, etc. if the source directory doesn't exist.

Reply:
I think the basenames carry much more information than dirnames do.

Reply:
Done

Reply:
FYI. The package installed with `~examples` shows the resulting message for each of the tests.